cmc-rnaseq-de/MSSM-Penn-Pitt_ACC_DLPFC_FEATURECOUNTS_GRCh38_CQN_Sparse.Rmd

```diff
@@ -742,30 +742,32 @@ Clinical, technical and batch variables selected after phase I, II and III of mo
 writeLines(paste('Using following covariates in the model:',
                  paste(adjust.covars, collapse=', '),
                  'as fixed effects and Individual_ID is chosen as random effect'))
-  
-# Post adjusted design matrix
-DM1 = getDesignMatrix(COVARIATES[,adjust.covars,drop=F],Intercept = F)
-DM1$design = DM1$design[,linColumnFinder(DM1$design)$indepCols]
-  
-# Estimate voom weights
+
+# Post adjusted formula
+random_effect <- "Individual_ID"
+formula <- glue::glue("~ ", glue::glue_collapse(adjust.covars, sep = " + ")) %>% 
+  glue::glue(" + (1|{random_effect})")
+
+# Estimate voom weights with DREAM
 tmp = NEW.COUNTS
 tmp[is.na(tmp)] = 0
-VOOM.GENE_EXPRESSION = voom(tmp, design=DM1$design, plot=F)
+geneExpr = DGEList(tmp)
+geneExpr = edgeR::calcNormFactors(geneExpr)
 
-# Estimate consensus correlation coefficient
-correlation <- parallelDuplicateCorrelation(VOOM.GENE_EXPRESSION, block = COVARIATES$Individual_ID)
-  
-# Re-calculate voom weights with correlation of random effects
-VOOM.GENE_EXPRESSION = voom(tmp, 
-                            design=DM1$design, plot=F,
-                            block = COVARIATES$Individual_ID, 
-                            correlation = correlation$cor)
+VOOM.GENE_EXPRESSION = variancePartition::voomWithDreamWeights(counts = tmp, 
+                                                               formula = formula,
+                                                               data = COVARIATES )
 
 # Fit linear model using new weights and new design
-VOOM.GENE_EXPRESSION$E = CQN.GENE_EXPRESSION$E
-ADJUSTED.FIT = lmFit(VOOM.GENE_EXPRESSION)
-  
-# Residuals after normalisation
+# CQN NA counts to 0 counts for input to dream()
+tmp = CQN.GENE_EXPRESSION$E
+tmp[is.na(tmp)] = 0
+VOOM.GENE_EXPRESSION$E = tmp
+ADJUSTED.FIT = variancePartition::dream(exprObj = VOOM.GENE_EXPRESSION, 
+                                        formula = formula,
+                                        data = COVARIATES)
+
+# Residuals after normalisation - residuals.MArrayLM() does not take MArrayLM2 objects
 RESIDUAL.GENE_EXPRESSION = residuals.MArrayLM(ADJUSTED.FIT, CQN.GENE_EXPRESSION$E)
```

Comment:
- `residuals.MArrayLM()` does not take MArrayLM2 objects as input. @GabrielHoffman do you have a strategy for transforming this object or alternative to computing residuals?

Reply:
Great suggestion.  I have add v1.17.3:
- dream(..., computeResiduals=TRUE) now computes residuals and allows use of residuals() function

So it will work with the new version

Reply:
v1.17.3 is installed and when I execute `dream()` with the additional argument `computeResiduals = TRUE`, I return `extra argument(s) ‘computeResiduals’ disregardedextra argument(s) ‘computeResiduals’ disregardedDividing work into 100 chunks...`